src/ASM/ASMs3DLag.C

```diff
@@ -930,18 +930,38 @@ bool ASMs3DLag::evalSolution (Matrix& sField, const Vector& locSol,
 int ASMs3DLag::findElement(double u, double v, double w,
                            double* xi, double* eta, double* zeta) const
 {
-  int elmx = std::min(nx-2.0, floor(u*(nx-1)));
-  int elmy = std::min(ny-2.0, floor(v*(ny-1)));
-  int elmz = std::min(nz-2.0, floor(w*(nz-1)));
+  const std::array<std::pair<double,int>,3> knot {{
+    {u, svol->basis(0).knotInterval(u)},
+    {v, svol->basis(1).knotInterval(v)},
+    {w, svol->basis(2).knotInterval(w)}
+  }};
+
+  const std::array<int,3> elm {
+    std::get<1>(knot[0]) - (p1 - 1),
+    std::get<1>(knot[1]) - (p2 - 1),
+    std::get<1>(knot[2]) - (p3 - 1)
```

Comment:
Is it better to use `std::get<1>()` instead of just `.second` when the argument is a `std::pair`? Thought `get`was indented for `std::tuple` which can have more than two values.

Reply:
not really better, just habitual because it makes code more generic. changed to .first / .second.